pkg/proxy/apis/kubeproxyconfig/validation/validation.go

```diff
@@ -238,3 +240,30 @@ func validateIPVSSchedulerMethod(scheduler kubeproxyconfig.IPVSSchedulerMethod,
 	}
 	return allErrs
 }
+
+func validateKubeProxyNodePortAddress(nodePortAddresses []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+LOOP:
+	for i := range nodePortAddresses {
+		if nodePortAddresses[i] == string(kubeproxyconfig.DefaultRoute) {
+			// special valid cases
+			continue LOOP
+		}
+		if nodePortAddresses[i] == "" {
+			allErrs = append(allErrs, field.Invalid(fldPath, nodePortAddresses, "must be non-empty"))
+			break
+		}
+		if _, _, err := net.ParseCIDR(nodePortAddresses[i]); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath, nodePortAddresses, "must be a valid IP block"))
+			break
+		}
+		if nodePortAddresses[i] == string(kubeproxyconfig.IPv4ZeroCIDR) || nodePortAddresses[i] == string(kubeproxyconfig.IPv6ZeroCIDR) {
```

Comment:
why bother with this?  It's not going to make any difference, and it will break when we introduce dual-stack support.

Reply:
Fixed now.